wicker/core/storage.py

```diff
@@ -275,24 +275,47 @@ class WickerPathFactory:
                     - _SUCCESS
                     - part-0-attempt-1234.parquet
                     - part-1-attempt-2345.parquet
+
+    When a prefix_replace_path is passed the paths of each location are altered such that
+    the prefix_to_trim is replaced with prefix_replace_path. This is generally done in the case
+    of mounted buckets where the fully-qualified posix path is required.
     """
 
-    def __init__(self, root_path: str, store_concatenated_bytes_files_in_dataset: bool = False) -> None:
+    def __init__(
+        self, root_path: str, prefix_replace_path: str = "", store_concatenated_bytes_files_in_dataset: bool = False
```

Comment:
I preferred empty string here because in `__format_pathing` I want to do the replace without having another conditional. It's more straight forward to me at least that we should default to empty  and have this default to empty so that we don't need another conditional to replace in that function.